package/MDAnalysis/lib/distances.py

```diff
@@ -24,19 +24,60 @@
 Functions
 ---------
 
-.. autofunction:: distance_array(reference, configuation [, box [,result]])
-.. autofunction:: self_distance_array(reference [, box [,result]])
-.. autofunction:: calc_bonds(atom1, atom2 [, box, [,result]])
-.. autofunction:: calc_angles(atom1, atom2, atom3 [,box [, result]])
-.. autofunction:: calc_dihedrals(atom1, atom2, atom3, atom4 [,box [, result]])
-.. autofunction:: apply_PBC(coordinates, box)
-.. autofunction:: transform_RtoS(coordinates, box)
-.. autofunction:: transform_StoR(coordinates, box)
+All functions take the optional keyword *mode*, which determines the
+type of acceleration. Currently, the following choices are implemented:
+
+======== ======================== ======================================
+*mode*   module                   description
+======== ======================== ======================================
+"serial" :mod:`_distances`        serial implementation in C/Cython
+
+"OpenMP" :mod:`_distances_openmp` parallel implementation in C/Cython
+                                  with OpenMP
+======== ======================== ======================================
+
+.. autofunction:: distance_array(reference, configuration [, box [, result [, mode]]])
+.. autofunction:: self_distance_array(reference [, box [,result [, mode]]])
+.. autofunction:: calc_bonds(atom1, atom2 [, box, [, result [, mode]]])
+.. autofunction:: calc_angles(atom1, atom2, atom3 [,box [, result [, mode]]])
+.. autofunction:: calc_dihedrals(atom1, atom2, atom3, atom4 [,box [, result [, mode]]])
+.. autofunction:: apply_PBC(coordinates, box [, mode])
+.. autofunction:: transform_RtoS(coordinates, box [, mode])
+.. autofunction:: transform_StoR(coordinates, box [,mode])
 """
 import numpy as np
 from numpy.lib.utils import deprecate
 
 from .mdamath import triclinic_vectors, triclinic_box
+
+
+# hack to select backend with mode=<mode> kwarg. Note that
+# the cython parallel code (prange) in paralle.distances is
+# independent from the OpenMP code
+import importlib
+_distances = {}
+_distances['serial'] = importlib.import_module("._distances",
```

Comment:
With these different backends, is it possible to cache the available functions when they are first imported?  So each "module" is a dict of functions